File: repos_estudiantes_ALVAREZ_primera_entrega/CALLE_CASTILLO_CRISTHIAN_JAVIER/PRACTICA1-PROGRAMACION-II/PRACTICA INF-121/PROGRAMACION ORIENTADA A OBJETOS/EJERCICIO 1/Celular.py
class Aplicacion:
    def __init__(self, nombre, tamaño):
        self.nombre = nombre
        self.tamaño = tamaño  # en MB

class Celular:
    def __init__(self):
        self.aplicaciones = []
        self.espacio_disponible = 1024  # en MB
        self.bateria = 100.0  

    def instalar_aplicacion(self, nombre, tamaño):
        if len(self.aplicaciones) >= 20:
            print("No se pueden instalar más de 20 aplicaciones.")
        elif tamaño > self.espacio_disponible:
            print("No hay suficiente espacio para instalar esta aplicación.")
        else:
            self.aplicaciones.append(Aplicacion(nombre, tamaño))
            self.espacio_disponible -= tamaño
            print(f"Aplicación '{nombre}' instalada exitosamente.")

    def usar_aplicacion(self, nombre, minutos):
        if self.bateria <= 0:
            print("Celular apagado")
            return

        app = next((a for a in self.aplicaciones if a.nombre == nombre), None)
        if not app:
            print(f"La aplicación '{nombre}' no está instalada.")
            return

        if app.tamaño > 250:
            consumo = 5 * (minutos / 10)
        elif app.tamaño > 100:
            consumo = 2 * (minutos / 10)
        else:
            consumo = 1 * (minutos / 10)

        self.bateria -= consumo
        if self.bateria < 0:
            self.bateria = 0

        print(f"Usaste '{nombre}' por {minutos} minutos. Batería actual: {self.bateria:.2f}%")
        if self.bateria == 0:
            print("Celular apagado")

    def mostrar_bateria(self):
        print(f"Batería restante: {self.bateria:.2f}%")

mi_celular = Celular()

mi_celular.instalar_aplicacion("Facebook", 300)
mi_celular.instalar_aplicacion("WhatsApp", 90)
mi_celular.instalar_aplicacion("YouTube", 250)

mi_celular.usar_aplicacion("Facebook", 30)   
mi_celular.usar_aplicacion("WhatsApp", 20)   
mi_celular.usar_aplicacion("YouTube", 40)    

mi_celular.mostrar_bateria()

File: repos_estudiantes_TARQUI_primera_entrega/MALLCU_CALANI_ALEJANDRO_GABRIEL/INTRODUCCION A POO/Ejercicio 7/Celular.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 */

package com.mycompany.celular;

/**
 *
 * @author USUARIO
 */
import java.util.HashMap;

public class Celular {
    private int espacioTotal;
    private int espacioDisponible;
    private int bateria;
    private HashMap<String, Integer> aplicaciones;

    // Constructor
    public Celular(int espacioTotal, int bateria) {
        this.espacioTotal = espacioTotal;
        this.espacioDisponible = espacioTotal;
        this.bateria = bateria;
        this.aplicaciones = new HashMap<>();
    }

    // Método para instalar aplicaciones
    public String instalarAplicacion(String nombre, int tamaño) {
        if (tamaño > espacioDisponible) {
            return "No hay suficiente espacio para instalar " + nombre + ".";
        }
        aplicaciones.put(nombre, tamaño);
        espacioDisponible -= tamaño;
        return "Aplicación " + nombre + " instalada con éxito.";
    }

    // Método para utilizar aplicaciones
    public String utilizarAplicacion(String nombre, int minutos) {
        if (!aplicaciones.containsKey(nombre)) {
            return "La aplicación " + nombre + " no está instalada.";
        }
        int tamaño = aplicaciones.get(nombre);
        double consumo;

        if (tamaño > 250) {
            consumo = (5.0 * minutos) / 10;
        } else if (tamaño > 100) {
            consumo = (2.0 * minutos) / 10;
        } else {
            consumo = (1.0 * minutos) / 10;
        }

        if (bateria - consumo <= 0) {
            bateria = 0;
            return "Celular apagado por batería agotada.";
        }
        bateria -= consumo;
        return "Usaste " + nombre + " por " + minutos + " minutos. Batería restante: " + bateria + "%.";
    }

    // Método para mostrar el porcentaje de batería
    public String mostrarBateria() {
        return "Batería restante: " + bateria + "%";
    }

    // Pruebas de la clase
    public static void main(String[] args) {
        Celular celular = new Celular(1024, 100);

        System.out.println(celular.instalarAplicacion("WhatsApp", 50));
        System.out.println(celular.instalarAplicacion("YouTube", 300));
        System.out.println(celular.utilizarAplicacion("YouTube", 20));
        System.out.println(celular.mostrarBateria());
        System.out.println(celular.utilizarAplicacion("WhatsApp", 50));
        System.out.println(celular.utilizarAplicacion("Spotify", 40)); // Aplicación no instalada
    }
}
